### PR TITLE
Update displaying of public lessons/sections

### PR DESCRIPTION
**Pull request by chrisjoywebb**

### CODE REVIEW

File: app/views/courses/index.html.erb [hunk 1]
```diff
@@ -14,6 +14,14 @@
...
+  <p><%= link_to 'New lesson', new_lesson_path, :class => "btn btn-success btn-large" %></p>
+<% end %>
+
 <% if can? :update, Lesson %>
   <p><%= link_to 'View deleted lessons', lessons_path + '?deleted=true' %></p>
 <% end %>
+
+<% if user_signed_in? %>
+  <%= link_to('Log out', destroy_user_session_path, method: :delete, class: 'btn btn-small btn-info') %>
+<% end %>
```
Comment (michaelrkn) [inline app/views/courses/index.html.erb]: navbar?

File: app/views/courses/index.html.erb [hunk 2]
```diff
@@ -5,7 +5,7 @@
 <%= render 'lessons/search' %>
 
 <ul class="unstyled-list">
-  <% @courses.each do |course| %>
+  <% public_courses_helper(@courses) do |course| %>
```
Comment (michaelrkn) [inline app/views/courses/index.html.erb]: move to controller

File: app/views/courses/show.html.erb [hunk 3]
```diff
@@ -18,7 +18,7 @@
           </ul>
         <% else %>
           <ul>
-            <%= render @course.sections %>
+            <%= render @course.sections.only_public %>
```
Comment (michaelrkn) [inline app/views/courses/show.html.erb]: accessible_by(current_user)

Comment (chrisjoywebb) [inline app/views/courses/show.html.erb]: done

File: app/views/courses/index.html.erb (hunk 1 above)
Comment (chrisjoywebb) [inline app/views/courses/index.html.erb]: done

**Pull request merged**